src/NumericalAlgorithms/Interpolation/InterpolationTargetKerrHorizon.hpp

```diff
@@ -0,0 +1,159 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+#include "ApparentHorizons/Strahlkorper.hpp"
+#include "ApparentHorizons/Tags.hpp"
+#include "ApparentHorizons/YlmSpherepack.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "NumericalAlgorithms/Interpolation/SendPointsToInterpolator.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrHorizon.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+namespace db {
+template <typename TagsList>
+class DataBox;
+}  // namespace db
+namespace intrp {
+namespace Tags {
+template <typename Metavariables>
+struct TemporalIds;
+}  // namespace Tags
+}  // namespace intrp
+/// \endcond
+
+namespace intrp {
+
+namespace OptionHolders {
+/// A surface that conforms to the horizon of a Kerr black hole in
+/// Kerr-Schild coordinates.
+struct KerrHorizon {
+  struct Lmax {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "KerrHorizon is expanded in Ylms up to l=Lmax"};
+  };
+  struct Center {
+    using type = std::array<double, 3>;
+    static constexpr OptionString help = {"Center of black hole"};
+  };
+  struct Mass {
+    using type = double;
+    static constexpr OptionString help = {"Mass of black hole"};
+  };
+  struct DimensionlessSpin {
+    using type = std::array<double, 3>;
+    static constexpr OptionString help = {"Dimensionless spin of black hole"};
+  };
+  using options = tmpl::list<Lmax, Center, Mass, DimensionlessSpin>;
+  static constexpr OptionString help = {
+      "A Strahlkorper conforming to the horizon (in Kerr-Schild coordinates)"
+      " of a Kerr black hole with a specified center, mass, and spin."};
+
+  KerrHorizon(size_t l_max_in, std::array<double, 3> center_in, double mass_in,
+              std::array<double, 3> dimensionless_spin_in,
+              const OptionContext& context = {});
+
+  KerrHorizon() = default;
+  KerrHorizon(const KerrHorizon& /*rhs*/) = default;
+  KerrHorizon& operator=(const KerrHorizon& /*rhs*/) = default;
+  KerrHorizon(KerrHorizon&& /*rhs*/) noexcept = default;
+  KerrHorizon& operator=(KerrHorizon&& /*rhs*/) noexcept = default;
+  ~KerrHorizon() = default;
+
+  // clang-tidy non-const reference pointer.
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+  size_t l_max{};
+  std::array<double, 3> center{};
+  double mass{};
+  std::array<double, 3> dimensionless_spin{};
+};
+
+bool operator==(const KerrHorizon& lhs, const KerrHorizon& rhs) noexcept;
```

Comment:
Not sure if you want to add `operator!=` for symmetry.

Reply:
Ok.  Should it be inlined?

Reply:
I'm fine with not inlining it since it won't be called often, if ever, outside of tests.